tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
@dorugeber to clarify my comment about the impact on performance: basically, if you have a loop, it is always better that accesses are adjacent in memory (spatial locality), e.g. A[i], A[i+1], A[i+2] etc. If you have jumps you may well screw a few things in the fetch-decode-execute pipeline (including vectorisation).  Essentially, if we sacrifice this property, it has to be for a good cause; for example, saving on a significant amount of operations on zeros. A single column (unless a very, very big one) is not a significant amount. Does this help ?